trail/upload/process_uploads/processors/decam_processor.py

```diff
@@ -0,0 +1,360 @@
+                    exts.append(hdu)
+
+        return exts
+
+    @classmethod
+    def canProcess(cls, uploadedFile, returnHdulist=False):
+        canProcess, hdulist = super().canProcess(uploadedFile, returnHdulist=True)
+
+        # Data examples I have seen Community Pipeines exclusively utilize the
+        # CompImageHDU headers and at any time in history there was at most 1
+        # Here, we bet that if we are near 62 CCDs encoded as CompImageHDUs,
+        # ignoring guider and focus, we are looking at DECam CP product
+        exts = cls._getScienceImages(hdulist)
+
+        if len(exts) > 60 and len(exts) <= 62:
+            canProcess = canProcess and True
+        else:
+            canProcess = canProcess and False
+
+        if returnHdulist:
+            return canProcess, hdulist
+        return canProcess
+
+    @classmethod
+    def normalizeImage(cls, image):
+        # astropy equalizer averages 4.3 seconds, the PIL approach can bring
+        # that down to cca 0.13. Without normalization the time is 0.04s
+        avg, std = image.mean(), image.std()
+        image[image>(avg + 0.5*std)] = avg + 0.5*std
+        image[image<(avg - 0.5*std)] = avg - 0.5*std
+        image = image/image.max()
+        image = (image*255).astype(np.uint8)
+
+        image = Image.fromarray(image, "L")
+        image = ImageOps.equalize(image, mask=None)
+
+        return image
+
+    def _createFocalPlaneImage(self, focalPlane):
+        for ext in self.exts:
+            # no matter how painful this is, if we don't, normalize will mutate
+            # in science data in place....
+            image = ext.data.copy()
+            image = self.normalizeImage(image)
+
+            # TODO: test here if the step-vise resizing is faster...
+            image = image.resize((focalPlane.scaledY, focalPlane.scaledX),
+                                 Image.ANTIALIAS)
+
+            focalPlane.add_image(image, ext.header["DETPOS"])
+
+        return focalPlane
+
+    def storeThumbnails(self, scaling=(4, 10)):
+
+        xdim = self.exts[0].header["NAXIS2"]
+        ydim = self.exts[0].header["NAXIS1"]
+        largePlane = DecamFocalPlane(scaling[0], (xdim, ydim))
+        smallPlane = DecamFocalPlane(scaling[1], (xdim, ydim))
+
+        # TODO: a note to fix os.path dependency when transitioning to S3
+        # and fix saving method from plt to boto3
+        smallPath = os.path.join(self.media_root, self.uploadedFile.basename+'_plane_small.jpg')
+        largePath = os.path.join(self.media_root, self.uploadedFile.basename+'_plane_large.jpg')
+
+        smallThumb = self._createFocalPlaneImage(smallPlane)
+        plt.imsave(smallPath, smallThumb.planeImage.T, pil_kwargs={"quality":30})
+        # due to the size of these images immediately release memory
+        del smallThumb
+
+        largeThumb = self._createFocalPlaneImage(largePlane)
+        plt.imsave(largePath, largeThumb.planeImage.T, pil_kwargs={"quality":30})
```

Comment:
Policy for file/image overwriting error-handling situations?

Reply:
See the comment about error handling [above](https://github.com/dirac-institute/trailblazer/pull/10#discussion_r654729848).

Is overwriting an error or not? I'm not sure in what cases it should be allowed, or should updating data be allowed only by manual cranking it. 
Is processing a file that produces the same standardized metadata an error - our standardized metadata is basically the observer location, observation time and observing instrument so this is not really sufficient to tell (f.e. DECam via Rubin will even have the same observing times but we can't tell they're the same pointing, different ccd at the moment) so we need to check if WCS matches I guess. Then I guess if WCS and metadata matches do we want to soft delete, hard delete or substitute in-place? Anyhow... 

The newest code centralizes the place were we save the images, in preparation for the move to S3. I assume we can make a unique string (read as: randomly generated string large enough to give an asymptotic guarantee of never duplicating) and centralizing that functionality for all processors too, as a quick fix, but there's a good chance this behavior will become clear once we answer above questions so I'm not particularly keen on resolving it in this PR. 

For this PR I've assumed no uploaded file ever will break any of the code in it in non-trivial ways.

Reply:
Yeah I'm not sure either 😄 we need a duplicates policy in general. If all the metadata (including WCS) matches, I think we'd want to default to not accepting the upload so we don't waste our upload/download cloud $ on duplicates. Does that sound reasonable?

Reply:
This sounds perfectly reasonable to me in this case. I just didn't have the capacity to think about all of the details here. To boot it felt like too much work that will leave more people pending on the merge of this PR.
Punted to issue https://github.com/dirac-institute/trailblazer/issues/15